### PR TITLE
wasm: Update proxy-wasm-cpp-host

### PR DESCRIPTION
**Pull request by mathetake**
Signed-off-by: mathetake <takeshi@tetrate.io>

ref: https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/98

### CODE REVIEW

**Comment by junr03** [comment]
Let us know when this is ready by changing with "ready for review"

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -897,10 +897,10 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "WebAssembly for Proxies (C++ host implementation)",
         project_desc = "WebAssembly for Proxies (C++ host implementation)",
         project_url = "https://github.com/proxy-wasm/proxy-wasm-cpp-host",
-        version = "15827110ac35fdac9abdb6b05d04ee7ee2044dae",
-        sha256 = "77a2671205eb0973bee375a1bee4099edef991350433981f6e3508780318117d",
+        version = "53facbe7bdf25a81a88821629a678be2b8bb710b",
+        sha256 = "2426bc2725ddd0c982dee723179858f7726828cdc7d44655d09584c64224e8d7",
         strip_prefix = "proxy-wasm-cpp-host-{version}",
-        urls = ["https://github.com/proxy-wasm/proxy-wasm-cpp-host/archive/{version}.tar.gz"],
+        urls = ["https://github.com/mathetake/proxy-wasm-cpp-host/archive/{version}.tar.gz"],
```
Comment (moderation) [inline bazel/repository_locations.bzl:903]: Is the plan here to really change to a fork of the upstream `proxy-wasm/proxy-wasm-cpp-host` repo? From a deps point of view it is preferable to make the changes required to the upstream

**Review by moderation — CHANGES_REQUESTED** [review]

File: bazel/repository_locations.bzl (hunk 1 above)
Comment (mathetake) [reply bazel/repository_locations.bzl:903]: I would like you to know that if we at [proxy-wasm-cpp-host](https://github.com/proxy-wasm/proxy-wasm-cpp-host) folks (mostly me and @PiotrSikora) change the repo pointing to to our folk repositories, that means we just want to run the CI in Envoy to test them against Envoy's test cases, and I NO intention to actually change the url to our folk repositories.

Anyway, thanks @moderation always for kind review.

Comment (PiotrSikora) [reply bazel/repository_locations.bzl:903]: Yeah, we only change this temporarily to perform end-to-end tests with Envoy before the change is merged upstream.

Note that we always try to leave the PR in Envoy as a draft until the change is merged there, and re-point it to merged commit before marking it as ready for review.

**Comment by mathetake** [comment]
As I mentioned in https://github.com/envoyproxy/envoy/pull/14056#discussion_r525618727, the initial purpose is to test the proxy-wasm's PR https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/98 and is achieved. So I'm closing

**Pull request closed without merging**